### ISSUE TITLE
no-explicit-generics doesn't work correctly

### ISSUE THREAD

**Issue by mark-akturatech** [JasonWeinzierl/eslint-plugin-rxjs-x#77]
version: 0.6.0

Gday,
First - great job. Thank you.

Not sure if worth removing `no-explicit-generics` from strict rule set.

 The rule is generating an error for lines of code like this...
 ```
private readonly itemSelectionUpdateSubject$ = new BehaviorSubject<ItemSelectionModel | undefined>(undefined);
```

If i follow the rule and remove the generic then when the subject is set it will error as the subject has type 'undefined'.

Maybe an option to ignore undefined could work here or probably would be better if the rule ignored generics with union types (or had an option to enable that behavior).

**Comment by Dmytro27Ind** [comment]
I encountered a similar issue where ESLint reports the following error:
 `ESLint: Explicit generic type arguments are forbidden.(rxjs-x/ no-explicit-generics)`

Here's the original code that triggered the error:

```ts
private itemsLoadedSubject$ = new BehaviorSubject<ListItem | null>(null);
```

However, I couldn't remove the explicit `ListItem` type because, without it, the inferred type defaults to `null`.

To resolve this, I updated the code as follows:

```ts
private itemsLoadedSubject$ = new BehaviorSubject<ListItem | null>(null); // eslint-disable-line rxjs-x/no-explicit-generics
```

(version 0.6.0)

**Comment by Dmytro27Ind** [comment]
### Proposed Solution

Introduce a configuration option for the rule to allow explicit generics specifically for union types. For example:

```ts
"rxjs-x/no-explicit-generics": [ "error", { "allowUnionTypes": true } ]
```

**Comment by JasonWeinzierl** [comment]
It looks like this is a known issue in the upstream repos:
- https://github.com/cartant/rxjs-tslint-rules/issues/95
- https://github.com/cartant/eslint-plugin-rxjs/issues/102

typescript-eslint itself may add support for this in the future: https://github.com/typescript-eslint/typescript-eslint/issues/4978 .  If that issue gets resolved, we may be able remove `no-explicit-generics` entirely.

But for the time being, I agree this should be removed from the  `strict` configuration until this bug is resolved.  I'll be releasing v0.6.1 shortly with this rule removed from `strict`.

**Comment by rdamazio** [comment]
Thanks for all the work on this plugin btw!

Is it sufficient to check if the generic type exactly matches the default value type? (so e.g. a union type wouldn't exactly match, and thus it wouldn't warn then)

**Comment by JasonWeinzierl** [comment]
This is fixed in v0.7.4

**Comment by JasonWeinzierl** [comment]
Ah I see.  The fix released for v0.7.4 only addressed union types declared inline in the subject's generic type... not union types or enums declared elsewhere.  Reopening.
